pkg/multicluster/register/registrant.go

```diff
@@ -630,17 +628,15 @@ func (c *clusterRegistrant) getTokenForSa(
 // issued for the domain contained in the value of `--local-cluster-domain-override`.
 //
 // this function call is a no-op if those conditions are not met
-func (c *clusterRegistrant) hackClusterConfigForLocalTestingInKIND(
+func (c *clusterRegistrant) hackClusterConfigForLocalTesting(
 	remoteCluster *api.Cluster,
-	remoteContextName string,
 ) error {
 	serverUrl, err := url.Parse(remoteCluster.Server)
 	if err != nil {
 		return err
 	}
 
-	if strings.HasPrefix(remoteContextName, "kind-") &&
-		(serverUrl.Hostname() == "127.0.0.1" || serverUrl.Hostname() == "localhost") &&
+	if (serverUrl.Hostname() == "127.0.0.1" || serverUrl.Hostname() == "localhost") &&
```

Comment:
i think we should modify this so that we always respect `c.localAPIServerAddress` if it is set; there are use cases where the address a user uses to communicate with kube from their local machine may differ from the one used by pods running in the cluster.

so i think what that would mean is to separate out the following:

```go
if c.localAPIServerAddress {
	if host, newPort, err := net.SplitHostPort(c.localAPIServerAddress); err == nil {
			c.localAPIServerAddress = host
			port = newPort
		}
	}
	remoteCluster.Server = fmt.Sprintf("https://%s:%s", c.localAPIServerAddress, port)
}
```

as far as the other changes we do, we can comntiue to put them behind the check for 
`serverUrl.Hostname() == "127.0.0.1" || serverUrl.Hostname() == "localhost"`, though i think we might want to consider exposing flags for these as well (as there are other situations where a user may want to configure custom options for TLS (outside of using Kind/minikube)